subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/IsolatingIncrementalAnnotationProcessingIntegrationTest.groovy

```diff
@@ -114,11 +124,12 @@ class IsolatingIncrementalAnnotationProcessingIntegrationTest extends AbstractIn
         run "compileJava"
 
         then:
-        outputs.deletedClasses("A", "AHelper")
+        outputs.deletedFiles("A", "AHelper", "AHelperResource.txt")
```

Comment:
I think `deletedFiles` should not have magical handling for classes. Instead I'd have two method calls here, one for `deletedClasses` and one for `deletedResources`

Reply:
Okay.  I will make that change.  In https://github.com/gradle/gradle/blob/master/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CompilationOutputsFixture.groovy, you can see there's nothing really specific to classes in any of those methods; it's just removing file extensions.  But it's nice for test clarity to make believe that there's a difference :).

Reply:
Ah, there is a bit of a problem here.  All of the assertion functions in `CompilationOutputsFixture` currently check that _all_ of the deleted/changed classes/files match the arguments.  So, if we split this into two separate assertions, we'd need to do filtering within those functions (e.g. `deletedClasses` filters for only those things with a `.class` extension and `deletedResources` filters for the inverse of that).  But that change couldn't be done without changing other callers of deletedClasses that aren't even JVM based, like [SwiftIncrementalCompileIntegrationTest](https://github.com/gradle/gradle/blob/70704e6cafda84b0bcce19ed6e2c6850698cc0ca/subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftIncrementalCompileIntegrationTest.groovy#L128).

So I am a bit hesitant to make this change.

Reply:
Got it, leave it as it is then :)